vassal-app/src/main/java/VASSAL/tools/image/svg/SVGImageUtils.java

```diff
@@ -78,35 +91,108 @@ public static Dimension getImageSize(InputStream in) throws IOException {
    * @return the image dimensions
    * @throws IOException if the image cannot be read
    */
-  public static Dimension getImageSize(String name, InputStream in)
-                                                          throws IOException {
-    // get the SVG
-    final Document doc;
-    try (in) {
-      synchronized (factory) {
-        doc = factory.createDocument(null, in);
-      }
+  public static Dimension getImageSize(String file, InputStream in) throws IOException {
+    try {
+      return getImageSize(getDocument(file, in));
     }
     catch (FileNotFoundException e) {
-      throw new ImageNotFoundException(name, e);
+      throw new ImageNotFoundException(file, e);
     }
     catch (DOMException | IOException e) {
-      throw new ImageIOException(name, e);
+      throw new ImageIOException(file, e);
     }
+  }
 
-    // get the default image width and height
-    final Element root = doc.getDocumentElement();
-    try {
-      final int width = (int) (Float.parseFloat(
-        root.getAttributeNS(null, "width").replaceFirst("px", "")) + 0.5);
-      final int height = (int) (Float.parseFloat(
-        root.getAttributeNS(null, "height").replaceFirst("px", "")) + 0.5);
+  public static Dimension getImageSize(SVGDocument doc) throws IOException {
+    final SVGSVGElement root = doc.getRootElement();
+    final String ws = root.getAttributeNS(null, SVGConstants.SVG_WIDTH_ATTRIBUTE);
+    final String hs = root.getAttributeNS(null, SVGConstants.SVG_HEIGHT_ATTRIBUTE);
+    final String vbs = root.getAttributeNS(null, SVGConstants.SVG_VIEW_BOX_ATTRIBUTE);
+
+    final BridgeContext bctx = new BridgeContext(new UserAgentAdapter());
+    final UnitProcessor.Context uctx = UnitProcessor.createContext(bctx, root); 
+
+    float w = -1.0f;
+    float h = -1.0f;
+    float[] vb = null;
+
+    // try to parse the width
+    if (!ws.isEmpty()) {
+      try {
+        w = UnitProcessor.svgHorizontalLengthToUserSpace(ws, SVGConstants.SVG_WIDTH_ATTRIBUTE, uctx);
+      }
+      catch (BridgeException e) {
+        // the width was invalid
+        throw new IOException(e);
+      }
+    }
 
-      return new Dimension(width, height);
+    // try to parse the height
+    if (!hs.isEmpty()) {
+      try {
+        h = UnitProcessor.svgVerticalLengthToUserSpace(hs, SVGConstants.SVG_HEIGHT_ATTRIBUTE, uctx);
+      }
+      catch (BridgeException e) {
+        // the height was invalid
+        throw new IOException(e);
```

Comment:
Note that the former "ImageIOException" I believe goes to a non-stack-dump "warning format", Does IOException? I would certainly recommend image-file loading not lead to stack-dump.

Reply:
If you look outward from where this particular overload of `getImageSize()` is called, you'll find the call is inside a catch which rethrows `IOException` wrapped in an `ImageIOException`. See lines 94-104 (new).